src/BaseNode.ts

```diff
@@ -13,11 +19,47 @@ export default class BaseNode {
     }
 
     public addChild(childNode: BaseNode) {
+        this.children = this.children || [];
         this.children.push(childNode);
     }
 
-    // TODO: change the method name
-    static matchClass(substr: string) {
-        return substr.length;
+    public isLeaf() {
+        return this.leaf;
+    }
+
+    public getChildren() {
+        return this.children;
+    }
+
+    public getData() {
+        return this.data;
+    }
+
+    print(stream: { write: (str: string) => void }, offset: number) {
```

Comment:
Will we use a standard `Node.js` streams or it doesn't matter in this case?

Reply:
That's for both process.stdout and writable treams